### ISSUE TITLE
Build procedure - skip cache folders, logs and uploads 

### ISSUE THREAD

**Issue by Kaik** [zikula/core#3950]
I'm not sure if it will help anyone but in case you are building the core from source and develop at the same time you might find this helpful

This still has a few issues it leaves mentioned folders only content is skipped and maybe there is a better way to do it... 

```
        // Will exclude everything under these directories

        $exclude = [
            'src/var/cache/dev',
            'src/var/cache/prod',
            'src/var/logs/dev.log',
            'src/var/logs/prod.log',
            // custom
            'src/web/uploads/news',
            'src/web/uploads/avatars',
            'src/web/uploads/classifieds',
            'src/web/uploads/smileys',
        ];

        /**
         * @param SplFileInfo $file
         * @param mixed $key
         * @param RecursiveCallbackFilterIterator $iterator
         * @return bool True if you need to recurse or if the item is acceptable
         */
        $filter = function ($file, $key, $iterator) use ($exclude) {
            $isDirExcluded = false;
            foreach ($exclude as $excludeDir) {
                $isDirExcluded = $file->getPath() == $excludeDir ? true : $isDirExcluded;
            }

            if ($isDirExcluded) {
                return false;
            }

            // files
            if ($file->isFile()) {
                foreach ($exclude as $excludeFile) {
//                    if (strpos($file->getPathname(), $excludeFile) !== false) {
                    if ($file->getPathname() == $excludeFile) {

                        return false;
                    }
                }
            }

            return true;
        };

        $innerIterator = new RecursiveDirectoryIterator(
            $sourceDir,
            RecursiveDirectoryIterator::SKIP_DOTS
        );

        $iterator = new RecursiveIteratorIterator(
            new RecursiveCallbackFilterIterator($innerIterator, $filter),
            \RecursiveIteratorIterator::SELF_FIRST
        );

        $filesystem->mirror($sourceDir, "$buildDir/$name", $iterator);

```

**Comment by Guite** [comment]
Shouldn't the go into a wiki page instead of an issue?

**Comment by Kaik** [comment]
I guess this can go there. I'm not sure I know how to edit/add to wiki :)

**Comment by Guite** [comment]
Closing this. Basically you can just use `.gitignore` to avoid that "content" gets into development / repositories.
Feel free to reopen/comment if I misunderstood.

**Comment by Kaik** [comment]
It is not because it is getting into repositories it is because all this "content" which is custom to environment is copied during build process into build artifacts... 

My dev environment is a complete copy of my website on production... when I want to test and deploy I do build then I rsync that build to test env and then deploy.

Without this all custom files from web/uploads all cache and logs are added to the build which in some cases can be a few gigabytes+

In short build process is copying unneeded files that in some cases can be huge (dev.log) uploads folder or cache folders.

**Comment by Guite** [comment]
Understood.

Such things are tightly coupled to an individual workflow.

It is also possible to have for example two Git branches (`staging` and `production`) and a CI server which deploys to a dev environment first and after the changes are merged into production to a prod environment.

This page also contains many interesting ideas: https://symfony.com/doc/current/deployment.html
